zuul-core/src/main/java/com/netflix/zuul/netty/server/OriginResponseReceiver.java

```diff
@@ -182,7 +189,14 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         if (msg instanceof HttpRequestMessage) {
             promise.addListener((future) -> {
                 if (!future.isSuccess()) {
-                    fireWriteError("request headers", future.cause(), ctx);
+                    Throwable cause = ctx.channel().attr(SSL_HANDSHAKE_UNSUCCESS_FROM_ORIGIN_THROWABLE).get();
+                    if (cause != null) {
+                        // Set the specific SSL handshake error if the handlers have already caught them
+                        ctx.channel().attr(SSL_HANDSHAKE_UNSUCCESS_FROM_ORIGIN_THROWABLE).set(null);
+                        fireWriteError("request headers", cause, ctx);
```

Comment:
In this case, it would be prudent to log the original `future.cause()` at a debug level, just to make sure it isn't lost.   In the SSL handshake failures, it _usually_ is logged elsewhere, but it isn't guaranteed.   Logging it here would help debugging live instances incase there is additional context in it.

Reply:
Sounds good.